File: python-scms/old/src/app/authentication.py
import bcrypt

class AuthenticationService:
    @staticmethod
    def hash_password(password: str) -> str:
        password_bytes = password.encode('utf-8')
        salt = bcrypt.gensalt()
        hashed = bcrypt.hashpw(password_bytes, salt)
        return hashed.decode('utf-8')

    @staticmethod
    def verify_password(stored_password: str, provided_password: str) -> bool:
        stored_bytes = stored_password.encode('utf-8')
        provided_bytes = provided_password.encode('utf-8')
        return bcrypt.checkpw(provided_bytes, stored_bytes)

File: java-scms/src/auth/Authentication.java
package auth;

import org.mindrot.jbcrypt.BCrypt;

public class Authentication {
    public static String encryptPassword(String password) {
        return BCrypt.hashpw(password, BCrypt.gensalt());
    }

    public static boolean checkPassword(String password, Map<String, String> row) {
        return BCrypt.checkpw(password, row.get('password'));
    }
}
